pkg/trait/logging.go

```diff
@@ -90,8 +90,12 @@ func (l loggingTrait) Apply(environment *Environment) error {
 		if util.IsTrue(l.JsonPrettyPrint) {
 			envvar.SetVal(&environment.EnvVars, envVarQuarkusLogConsoleJsonPrettyPrint, True)
 		}
-	} else if util.IsNilOrTrue(l.Color) {
-		envvar.SetVal(&environment.EnvVars, envVarQuarkusLogConsoleColor, True)
+	} else {
+		envvar.SetVal(&environment.EnvVars, envVarQuarkusLogConsoleJson, False)
```

Comment:
I wonder, should this be done only when the user has set the `json` property to false? In general I think it's desirable to inherit from the Quarkus default behaviour, and avoid forcing environment variables.

Reply:
In this case, no. That would cause the problem that @lburgazzoli raised.

Reply:
My understanding is that in @lburgazzoli case, he disables it explicitly with `trait.camel.apache.org/logging.json: "false"`, so that would work?

Reply:
I see your point and I think it makes sense. There is a risk, though. 

Right now we add the required library for logging only if the [json flat is true](https://github.com/apache/camel-k/blob/main/pkg/trait/logging.go#L76). 

If we change that, we risk getting into a situation where Quarkus changes the default logging behavior from plain text to json or some other way/mechanism and we are left without control. 

So, although this is very unlikely, I still think it would be better to be explicit in this particular case, as we are talking about a core feature of the integration.

Reply:
@astefanutti to be clear: I am also fine with your suggestion ... Although I think it's safer to be explicit here, I am not strongly opinionated towards what I suggested. As such, so if you do feel it's the best way ... I am fine with it too.

Reply:
I don0t have any strong opinion here and both solution are fine

Reply:
I don't have a strong opinion either 😀. I'd be slightly inclined toward not adding a particular case. I guess the overall question is should Camel K be responsible for defaulting Quarkus runtime configuration, or should it delegates to Quarkus. Until we have a consensus on what is the best approach, it's not a big deal to choose one or the other for a few particular cases, so either way is fine with me 👍🏼.

Reply:
@astefanutti @lburgazzoli thanks guys. So, let's go with this as a work-around for now. 

Antonin's raised a good point here and I think I can take a look at solving this when I cleanup/refactor for #2541. At the very least this gives us some time to think about it and consider the approach.